### PR TITLE
[15127] Make reply react on edit

### LINKED ISSUE #15172: Pinned message, Message in the quote, Pinned in the system message are not updated when they are edited
**Steps to reproduce:**
1. Go to any chat as User_A and User_B (for both users chat should be opened in this step)
2. Send the message
3. Pin the message
4. Reply to the message
5. Check the message as User_B

**Actual  result:**
The message is not updated in the following places:
- Pinned banner
- Quote
image

- Pinned in the system message
image

https://user-images.githubusercontent.com/52490791/220895944-db76c68a-0fff-4328-abdf-8e2728a1ee78.mp4

**Expected result:**
The message is updated in the following places:
- Pinned banner
- Quote
- Pinned in the system message

**Additional info:**
The messages are updated only after the chat is reopened

**ENV:** Nightly 23 Feb 2023

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #15172 

Make message reply quoted message change if the original message is edited

status: ready

### CODE REVIEW

**Review by ilmotta — APPROVED** [review]

File: src/status_im/ui2/screens/chat/messages/message.cljs [hunk 1]
```diff
@@ -128,13 +128,14 @@
...
-                                                  :deleted-for-me? deleted-for-me?
-                                                  :chat-id         chat-id)]
+  [{:keys [message-id chat-id]} _ pin?]
+  (let [{:keys [deleted? deleted-for-me?]
+         :as   quoted-message} (get @(re-frame/subscribe [:chats/chat-messages chat-id])
+                                    message-id)
+        reply                  (assoc quoted-message
+                                      :deleted?        deleted?
+                                      :deleted-for-me? deleted-for-me?
+                                      :chat-id         chat-id)]
```
Comment (Parveshdhull) [inline src/status_im/ui2/screens/chat/messages/message.cljs]: - redundant `assoc` for keys `:deleted?` and `:deleted?-for-me`
- Not sure, just curious, is quoted-message has chat-id?

Comment (Parveshdhull) [reply src/status_im/ui2/screens/chat/messages/message.cljs]: just checked, message has chat-id, whole assoc is unnecessary 
image

File: src/status_im/ui2/screens/chat/messages/message.cljs [hunk 2]
```diff
@@ -128,13 +128,14 @@
           (:parsed-text content)))
 
 (defn quoted-message
-  [{:keys [message-id chat-id]} reply pin?]
```
Comment (Parveshdhull) [inline src/status_im/ui2/screens/chat/messages/message.cljs:131]: If we are not using reply, can we just remove it from params

File: src/status_im/ui2/screens/chat/messages/message.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im/ui2/screens/chat/messages/message.cljs]: Removed, Thanks Parvesh

File: src/status_im/ui2/screens/chat/messages/message.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im/ui2/screens/chat/messages/message.cljs:131]: Removed it, Thanks Parvesh

**Review by Parveshdhull — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @ibrkhalil, thanks for the fix, quoted messages are updated correctly now. 
But the issue is also relevant for pinned messages, where the update still does not work. Could you clarify if this part should be fixed separately?
If that's the case, could you update the PR description to indicate that this partially fixes the bug, please? We'll keep it open until it's completely fixed. Thanks

**Comment by ibrkhalil** [comment]
> Hey @ibrkhalil, thanks for the fix, quoted messages are updated correctly now. But the issue is also relevant for pinned messages, where the update still does not work. Could you clarify if this part should be fixed separately? If that's the case, could you update the PR description to indicate that this partially fixes the bug, please? We'll keep it open until it's completely fixed. Thanks

Hey @qoqobolo, I can't introduce the bug in pinned messages .. Can you please list reproduction steps ?

**Comment by ibrkhalil** [comment]
> > Hey @ibrkhalil, thanks for the fix, quoted messages are updated correctly now. But the issue is also relevant for pinned messages, where the update still does not work. Could you clarify if this part should be fixed separately? If that's the case, could you update the PR description to indicate that this partially fixes the bug, please? We'll keep it open until it's completely fixed. Thanks
> 
> Hey @qoqobolo, I can't introduce the bug in pinned messages .. Can you please list reproduction steps ?

BTW, [This](https://github.com/status-im/status-mobile/pull/15190#issuecomment-1450208506) PR solves some editing updating issues

**Comment by qoqobolo** [comment]
> Can you please list reproduction steps ?

@ibrkhalil  sure:
1. UserA sends a message to UserB in 1-1 chat
2. UserA pins the message
3. UserA edits the message
4. UserB checks the pinned messages banner in the chat

In the screenshot you can see that the massage `Heey` has been edited to `Heey updated`. UserB can see this in the message itself, in the system message (`Ironclad Distant... pinned a message`), but it is not updated at the top of the chat view in the banner.  
Screenshot 2023-03-01 at 17 54 00

This particular case is also reproducible in https://github.com/status-im/status-mobile/pull/15190

@ibrkhalil, this probably happened from my changes a while back. I forgot to follow up with the second subscription so it updates on edit

**Comment by ibrkhalil** [comment]
> > Can you please list reproduction steps ?
> 
> @ibrkhalil sure:
> 
> 1. UserA sends a message to UserB in 1-1 chat
> 2. UserA pins the message
> 3. UserA edits the message
> 4. UserB checks the pinned messages banner in the chat
> 
> In the screenshot you can see that the massage `Heey` has been edited to `Heey updated`. UserB can see this in the message itself, in the system message (`Ironclad Distant... pinned a message`), but it is not updated at the top of the chat view in the banner. Screenshot 2023-03-01 at 17 54 00
> 
> This particular case is also reproducible in #15190
> 
> @ibrkhalil, this probably happened from my changes a while back. I forgot to follow up with the second subscription so it updates on edit

Can you please check if the last bug (Pinned banner one) can still be reproduced after my last commit here ?

**Comment by qoqobolo** [comment]
@ibrkhalil works perfectly now, thank you!
PR can be merged.

**Pull request merged**